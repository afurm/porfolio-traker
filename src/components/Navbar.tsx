import React from 'react';
import Link from 'next/link';

import { AppBar, Toolbar, Button, Box, Typography } from '@mui/material';

const Navbar: React.FC = () => {

  
  return (
    <AppBar position="static" sx={{ backgroundColor: '#282c34' }}>
      <Toolbar>
        <Typography variant="h6" sx={{ flexGrow: 1 }}>
          <Link href="/" passHref>
            <a style={{ color: 'white', textDecoration: 'none', fontWeight: 'bold' }}>Crypto Portfolio Tracker</a>
          </Link>
        </Typography>
        <Box>
          <Link href="/dashboard" passHref>
            <Button color="inherit" sx={{ marginRight: 1 }}>Dashboard</Button>
          </Link>
        </Box>
      </Toolbar>
    </AppBar>
  );
};

export default Navbar;