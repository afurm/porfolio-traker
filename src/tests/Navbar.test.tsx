import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from '@/components/Navbar';

describe('Navbar Component Snapshot Test', () => {
  it('renders correctly', () => {
    const { asFragment, getByText } = render(<Navbar />);

    // Ensure that the title and dashboard link are present
    expect(getByText('Crypto Portfolio Tracker')).toBeInTheDocument();
    expect(getByText('Dashboard')).toBeInTheDocument();

    // Create a snapshot to validate the rendering
    expect(asFragment()).toMatchSnapshot();
  });
});
