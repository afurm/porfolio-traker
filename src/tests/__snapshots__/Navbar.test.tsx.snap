// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Snapshot Test renders correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1ezlszh-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-173147z-MuiTypography-root"
      >
        <a
          href="/"
        />
        <a
          style="color: white; text-decoration: none; font-weight: bold;"
        >
          Crypto Portfolio Tracker
        </a>
      </h6>
      <div
        class="MuiBox-root css-0"
      >
        <a
          href="/dashboard"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-18ulbdb-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Dashboard
          </button>
        </a>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
